tests/TestServer.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Stripe;
+
+/**
+ * Methods for managing a test HTTP server.
+ *
+ * The test server works by spawning a new process to run PHP's "php -s" built-in test server.
+ *
+ * It is probably very easy to leak processes or files using this helper. This was a
+ * quick and dirty tool for building confidence in certain HTTP behaviors, but please
+ * feel free to delete if this begins to be flaky or cause trouble. Also think carefully
+ * before writing new tests that use this helper.
+ */
+trait TestServer
+{
+    /** @var mixed */
+    protected $serverProc;
+
+    /** @var mixed */
+    protected $serverStderr;
+
+    // No significance to this. I picked a random port that seemed close in
+    // value to `stripe-mock`'s standard 12111.
+    protected $serverPort = 12113;
+
+    /**
+     * Makes a directory in a temporary path containing only an `index.php` file with
+     * the specified content ($code).
+     *
+     * @param mixed $code
+     */
+    private function makeTemporaryServerDirectory($code)
+    {
+        $dir = \sys_get_temp_dir() . \DIRECTORY_SEPARATOR . 'stripe-php-test-server';
```

Comment:
Thoughts on adding an extra qualifier to this to make sure we don't accidentally clash throughout multiple test runs (eg. timestamp)? Can also help inspect the generated code to make sure all looks fine.

Reply:
Good thought - I'd like to solve the problem when it arises.